core/primitives/src/congestion_info.rs

```diff
@@ -57,18 +136,374 @@ impl Default for CongestionInfo {
 
 impl CongestionInfo {
     /// How much gas another shard can send to us in the next block.
-    pub fn outgoing_limit(&self, _sender_shard: ShardId) -> Gas {
-        todo!()
+    pub fn outgoing_limit(&self, sender_shard: ShardId) -> Gas {
+        match self {
+            CongestionInfo::V1(inner) => inner.outgoing_limit(sender_shard),
+        }
+    }
+
+    /// How much gas we accept for executing new transactions going to any
+    /// uncongested shards.
+    pub fn process_tx_limit(&self) -> Gas {
+        match self {
+            CongestionInfo::V1(inner) => inner.process_tx_limit(),
+        }
+    }
+
+    /// Whether we can accept new transaction with the receiver set to this shard.
+    pub fn shard_accepts_transactions(&self) -> bool {
+        match self {
+            CongestionInfo::V1(inner) => inner.shard_accepts_transactions(),
+        }
+    }
+
+    pub fn congestion_level(&self) -> f64 {
```

Comment:
nit: Can you add a small comment and mention that congestion level is in the [0, 1] range?